### PR TITLE
[datadog_logs_custom_destination] Introduce logs custom destination resource

### PR DESCRIPTION
**Pull request by Prepager**
Following the release of the [Logs Custom Destinations](https://docs.datadoghq.com/api/latest/logs-custom-destinations/) Public API, we want to provide a Terraform resource for it

Example usage
```tf
resource "datadog_logs_custom_destination" "http_basic_auth" {
  name                               = "http basic"
  query                              = "source:dummy"
  forward_tags                       = false
  forward_tags_restriction_list      = ["a", "b"]
  forward_tags_restriction_list_type = "ALLOW_LIST"

  http_destination {
    endpoint = "https://example.org"
    basic_auth {
      username = "username"
      password = "password"
    }
  }
}
```

### CODE REVIEW

File: docs/resources/logs_custom_destination.md [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+<!-- schema generated by tfplugindocs -->
+## Schema
+
+### Required
+
+- `name` (String) The custom destination name.
+
+### Optional
+
+- `elasticsearch_destination` (Block List) The Elasticsearch destination. (see [below for nested schema](#nestedblock--elasticsearch_destination))
```
Comment (drichards-87) [inline docs/resources/logs_custom_destination.md:40]:
```suggestion
- `elasticsearch_destination` (Block List) The Elasticsearch destination. (See [below for nested schema](#nestedblock--elasticsearch_destination).)
```

File: docs/resources/logs_custom_destination.md [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+- `enabled` (Boolean) Whether logs matching this custom destination should be forwarded or not.
+- `forward_tags` (Boolean) Whether tags from the forwarded logs should be forwarded or not.
+- `forward_tags_restriction_list` (List of String) List of [keys of tags](https://docs.datadoghq.com/getting_started/tagging/#define-tags) to be filtered.
```
Comment (drichards-87) [inline docs/resources/logs_custom_destination.md:43]:
```suggestion
- `forward_tags_restriction_list` (List of String) List of [tag keys](https://docs.datadoghq.com/getting_started/tagging/#define-tags) to be filtered.
```

File: docs/resources/logs_custom_destination.md [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+				An empty list represents no restriction is in place and either all or no tags will be
+				forwarded depending on `forward_tags_restriction_list_type` parameter.
+- `forward_tags_restriction_list_type` (String) How `forward_tags_restriction_list` parameter should be interpreted.
```
Comment (drichards-87) [inline docs/resources/logs_custom_destination.md:46]:
```suggestion
- `forward_tags_restriction_list_type` (String) How the `forward_tags_restriction_list` parameter should be interpreted.
```

File: docs/resources/logs_custom_destination.md [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+				If `ALLOW_LIST`, then only tags whose keys on the forwarded logs match the ones on the restriction list
+				are forwarded.
+				`BLOCK_LIST` works the opposite way. It does not forward the tags matching the ones on the list. Valid values are `ALLOW_LIST`, `BLOCK_LIST`.
+- `http_destination` (Block List) The HTTP destination. (see [below for nested schema](#nestedblock--http_destination))
```
Comment (drichards-87) [inline docs/resources/logs_custom_destination.md:50]:
```suggestion
- `http_destination` (Block List) The HTTP destination. (See [below for nested schema](#nestedblock--http_destination).)
```

File: docs/resources/logs_custom_destination.md [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+- `query` (String) The custom destination query filter. Logs matching this query are forwarded to the destination.
+- `splunk_destination` (Block List) The Splunk HTTP Event Collector (HEC) destination. (see [below for nested schema](#nestedblock--splunk_destination))
```
Comment (drichards-87) [inline docs/resources/logs_custom_destination.md:52]:
```suggestion
- `splunk_destination` (Block List) The Splunk HTTP Event Collector (HEC) destination. (See [below for nested schema](#nestedblock--splunk_destination).)
```

File: docs/resources/logs_custom_destination.md [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+### Read-Only
+
+- `id` (String) The ID of this resource.
+
+<a id="nestedblock--elasticsearch_destination"></a>
+### Nested Schema for `elasticsearch_destination`
+
+Required:
+
+- `endpoint` (String) The destination for which logs will be forwarded to. Must have HTTPS scheme and forwarding back to Datadog is not allowed.
```
Comment (drichards-87) [inline docs/resources/logs_custom_destination.md:63]:
```suggestion
- `endpoint` (String) The destination for which logs will be forwarded to. Must have HTTPS scheme. Forwarding back to Datadog is not allowed.
```

File: docs/resources/logs_custom_destination.md [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+- `index_name` (String) Name of the Elasticsearch index (must follow [Elasticsearch's criteria](https://www.elastic.co/guide/en/elasticsearch/reference/8.11/indices-create-index.html#indices-create-api-path-params)).
+
+Optional:
+
+- `basic_auth` (Block List) Basic access authentication. (see [below for nested schema](#nestedblock--elasticsearch_destination--basic_auth))
```
Comment (drichards-87) [inline docs/resources/logs_custom_destination.md:68]:
```suggestion
- `basic_auth` (Block List) Basic access authentication. (See [below for nested schema](#nestedblock--elasticsearch_destination--basic_auth).)
```

File: docs/resources/logs_custom_destination.md [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+- `username` (String, Sensitive) The username of the authentication. This field is not returned by the API.
+
+
+
+<a id="nestedblock--http_destination"></a>
+### Nested Schema for `http_destination`
+
+Required:
+
+- `endpoint` (String) The destination for which logs will be forwarded to. Must have HTTPS scheme and forwarding back to Datadog is not allowed.
```
Comment (drichards-87) [inline docs/resources/logs_custom_destination.md:94]:
```suggestion
- `endpoint` (String) The destination for which logs will be forwarded to. Must have HTTPS scheme. Forwarding back to Datadog is not allowed.
```

File: docs/resources/logs_custom_destination.md [hunk 9]
```diff
@@ -0,0 +1,135 @@
...
+
+Optional:
+
+- `basic_auth` (Block List) Basic access authentication. (see [below for nested schema](#nestedblock--http_destination--basic_auth))
```
Comment (drichards-87) [inline docs/resources/logs_custom_destination.md:98]:
```suggestion
- `basic_auth` (Block List) Basic access authentication. (See [below for nested schema](#nestedblock--http_destination--basic_auth).)
```

File: docs/resources/logs_custom_destination.md [hunk 10]
```diff
@@ -0,0 +1,135 @@
...
+- `custom_header_auth` (Block List) Custom header access authentication. (see [below for nested schema](#nestedblock--http_destination--custom_header_auth))
```
Comment (drichards-87) [inline docs/resources/logs_custom_destination.md:99]:
```suggestion
- `custom_header_auth` (Block List) Custom header access authentication. (See [below for nested schema](#nestedblock--http_destination--custom_header_auth).)
```

File: docs/resources/logs_custom_destination.md [hunk 11]
```diff
@@ -0,0 +1,135 @@
...
+
+
+
+<a id="nestedblock--splunk_destination"></a>
+### Nested Schema for `splunk_destination`
+
+Required:
+
+- `access_token` (String, Sensitive) Access token of the Splunk HTTP Event Collector. This field is not returned by the API.
+- `endpoint` (String) The destination for which logs will be forwarded to. Must have HTTPS scheme and forwarding back to Datadog is not allowed.
```
Comment (drichards-87) [inline docs/resources/logs_custom_destination.md:126]:
```suggestion
- `endpoint` (String) The destination for which logs will be forwarded to. Must have HTTPS scheme. Forwarding back to Datadog is not allowed.
```

File: docs/resources/logs_custom_destination.md [hunk 12]
```diff
@@ -0,0 +1,135 @@
...
+
+## Import
+
+Import is supported using the following syntax:
+
+```shell
+# Custom destinations can be imported using the destination ID. Beware auth credentials can not be imported.
```
Comment (drichards-87) [inline docs/resources/logs_custom_destination.md:133]:
```suggestion
# Custom destinations can be imported using the destination ID. Caution: auth credentials can not be imported.
```

**Review by drichards-87 — APPROVED** [review]
Left some small suggestions from Docs and approved the PR.

**Review by antonio-ramadas-dd — APPROVED** [review]
Approved from Logs. I don't have much experience with Go and it's my first time reviewing a Terraform provider implementation, but it looks good to me. Also, I know you tested it locally which gives me greater confidence. Nice work! 🚀

File: docs/resources/logs_custom_destination.md (hunk 1 above)
Comment (Prepager) [reply docs/resources/logs_custom_destination.md:40]: The `see [below for nested schema]` part is from the automatic documentation generation. Not something I can change.

**Review by antonio-ramadas-dd — APPROVED** [review]

**Review by drichards-87 — APPROVED** [review]

**Review by amaskara-dd — APPROVED** [review]

**Pull request merged**